cpp/include/raft/core/detail/macros.hpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifndef _RAFT_HAS_CUDA
+#if defined(__CUDACC__)
+#define _RAFT_HAS_CUDA __CUDACC__
+#endif
+#endif
+
+#ifndef _RAFT_HOST_DEVICE
+#if defined(_RAFT_HAS_CUDA)
+#define _RAFT_HOST_DEVICE __host__ __device__
```

Comment:
Would we consider separating this into `_RAFT_HOST` and `_RAFT_DEVICE`?

Reply:
I can definitely split these out but I'd prefer to do it in a follow-on only because I'm trying to orchestrate this merge w/ cugraph and cuopt (the changes break them downstream) and I'm hoping not to have to re-run CI if I can avoid it.